snuba/stateful_consumer/states/bootstrap.py

```diff
@@ -14,12 +22,60 @@ class BootstrapState(State[ConsumerStateCompletionEvent, ConsumerStateData]):
     control topic.
     """
 
+    def __init__(self,
+        topic: str,
+        bootstrap_servers: Sequence[str],
+        group_id: str,
+    ):
+        super(BootstrapState, self).__init__()
+
+        def on_partitions_assigned(
+            consumer: Consumer,
+            partitions: Sequence[TopicPartition],
+        ):
+            pass
+
+        def on_partitions_revoked(
+            consumer: Consumer,
+            partitions: Sequence[TopicPartition],
+        ):
+            pass
+
+        self.__consumer = StrictConsumer(
+            topic=topic,
+            bootstrap_servers=bootstrap_servers,
+            group_id=group_id,
+            auto_offset_reset="earliest",
+            partition_assignment_timeout=settings.SNAPSHOT_CONTROL_TOPIC_INIT_TIMEOUT,
+            on_partitions_assigned=on_partitions_assigned,
+            on_partitions_revoked=on_partitions_revoked,
+            on_message=self.__handle_msg,
+        )
+
+    def __handle_msg(self, message: Message) -> CommitDecision:
+        logger.info(
+            "MSG %r %r %r",
+            message,
```

Comment:
The Kafka `Message.__repr__` doesn't have any useful information in it, other than the memory address, it's probably not worth adding in here unless you really want the object identity for some reason.

I've been thinking it might be nice to wrap the `Consumer` class in our usage to add a message proxy or something that has a usable `__repr__` (and also type information!) since this is going to show up in pretty much any stack trace that is part of a Kafka consumer. I haven't got to this because it's a lot of work without too much reward, but I think it might be worth doing at some point. (Definitely not as part of this PR, though.)

Reply:
Oh well, this was just to test it worked. This line is gone in the PR above replaced by the actual logic.